test/AtlasVerification.t.sol

```diff
@@ -0,0 +1,676 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "forge-std/Test.sol";
+
+import { AtlasVerification } from "../src/contracts/atlas/AtlasVerification.sol";
+import { DAppConfig, DAppOperation, CallConfig } from "../src/contracts/types/DAppApprovalTypes.sol";
+import { UserOperation } from "../src/contracts/types/UserCallTypes.sol";
+import { SolverOperation } from "../src/contracts/types/SolverCallTypes.sol";
+import { ValidCallsResult } from "../src/contracts/types/ValidCallsTypes.sol";
+import { TxBuilder } from "../src/contracts/helpers/TxBuilder.sol";
+import { DummyDAppControl, CallConfigBuilder } from "./base/DummyDAppControl.sol";
+import { AtlasBaseTest } from "./base/AtlasBaseTest.t.sol";
+import { SimpleRFQSolver } from "./SwapIntent.t.sol";
+
+
+contract AtlasVerificationTest is AtlasBaseTest {
+    TxBuilder txBuilder;
+    DummyDAppControl dAppControl;
+    CallConfig callConfig;
+
+    struct Sig {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    function setUp() public virtual override {
+        // we reset the callConfig to all false for each test
+        // this allows us to selectively enable flags for each test.
+        // we then need to call refreshGlobals() to rebuild the
+        // dAppControl and txBuilder in the test itself. not every test
+        // needs to do this, so we try handle the most common case here.
+        callConfig = CallConfigBuilder.allFalseCallConfig();
+        refreshGlobals();
+    }
+
+    function refreshGlobals() public {
+        AtlasBaseTest.setUp();
+        dAppControl = buildDummyDAppControl();
+        txBuilder = buildTxBuilder();
+    }
+
+    function buildDummyDAppControl() public returns (DummyDAppControl) {
+        // Deploy new DummyDAppControl Controller from new gov and initialize in Atlas
+        vm.startPrank(governanceEOA);
+        DummyDAppControl control = new DummyDAppControl(address(atlas), governanceEOA, callConfig);
+        atlasVerification.initializeGovernance(address(control));
+        atlasVerification.integrateDApp(address(control));
+        vm.stopPrank();
+
+        return control;
+    }
```

Comment:
I did something similar for the EE tests: https://github.com/FastLane-Labs/atlas/blob/ee-tests/test/ExecutionEnvironment.t.sol#L40-L58

I think we can make that available at a higher level so it can be used in any test file.